Controller.hpp

```diff
@@ -18,35 +20,66 @@
 namespace Procon {
 
 	constexpr size_t exchangeLen{ 0x400 };
+
+	struct AccelData {
+		float x;
+		float y;
+		float z;
+	};
+
+	struct GyroData {
+		float x;
+		float y;
+		float z;
+	};
+
 	struct AxisRange {
-		uchar min;
-		uchar max;
+		uint16_t min;
+		uint16_t max;
 	};
+
 	struct StickRange {
 		AxisRange x;
 		AxisRange y;
 	};
+
 	struct StickPoint {
-		uchar x;
-		uchar y;
+		uint16_t x;
+		uint16_t y;
 	};
+
+	struct SensorCalibration {
+		std::array<float, 3> accelCoeff;
+		std::array<float, 3> gyroCoeff;
+	};
+
 	struct CalibrationData {
 		StickRange left;
 		StickRange right;
 		StickPoint leftCenter;
 		StickPoint rightCenter;
+		SensorCalibration motionSensor;
 	};
+
 	void SetDefaultCalibration(CalibrationData &dat);
+
 	struct HIDCloser {
 		void operator()(hid_device *ptr);
 	};
+
 	struct ExpandedPadState {
 		XINPUT_GAMEPAD xinState;
+		std::vector<std::tuple<Button, bool>> buttons;
```

Comment:
All buttons (except the Share button) were contained within the XINPUT_GAMEPAD struct. What was your reasoning for duplicating them here?